### PR TITLE
Reland "MTM: fix issues after cursory reading"

### PR DESCRIPTION
**Pull request by artagnon**
Change: Using ArrayRef{} constructor syntax causes a build failure on gcc-7. Hence, revert instances of this change.

### CODE REVIEW

**Comment by s-barannikov** [comment]
(passing by) What issues does it fix? I only see questionable style changes.

**Comment by artagnon** [comment]
> (passing by) What issues does it fix? I only see questionable style changes.

See #100404 for the original discussion: I initially marked it as a style change, but a reviewer requested dropping the NFC tag.

File: llvm/lib/CodeGen/MachineTraceMetrics.cpp [hunk 1]
```diff
@@ -133,7 +127,7 @@ MachineTraceMetrics::getResources(const MachineBasicBlock *MBB) {
 
   // Scale the resource cycles so they are comparable.
   unsigned PROffset = MBB->getNumber() * PRKinds;
-  for (unsigned K = 0; K != PRKinds; ++K)
+  for (unsigned K = 0; K < PRKinds; ++K)
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineTraceMetrics.cpp:130]: The coding standard doesn't specify which style of control condition is preferred, so there is no reason to change this line.

File: llvm/lib/CodeGen/MachineTraceMetrics.cpp [hunk 2]
```diff
@@ -546,7 +540,7 @@ MachineTraceMetrics::Ensemble::invalidate(const MachineBasicBlock *BadMBB) {
   if (BadTBI.hasValidHeight()) {
     BadTBI.invalidateHeight();
     WorkList.push_back(BadMBB);
-    do {
+    while (!WorkList.empty()) {
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineTraceMetrics.cpp:543]: No reason to change this either, especially when it is clear that the container is non-empty on entry.

File: llvm/lib/CodeGen/MachineTraceMetrics.cpp [hunk 3]
```diff
@@ -605,7 +599,7 @@ void MachineTraceMetrics::Ensemble::verify() const {
 #ifndef NDEBUG
   assert(BlockInfo.size() == MTM.MF->getNumBlockIDs() &&
          "Outdated BlockInfo size");
-  for (unsigned Num = 0, e = BlockInfo.size(); Num != e; ++Num) {
+  for (unsigned Num = 0; Num < BlockInfo.size(); ++Num) {
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineTraceMetrics.cpp:602]: The left form is [preferred](https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop).

File: llvm/lib/CodeGen/MachineTraceMetrics.cpp [hunk 4]
```diff
@@ -969,10 +963,8 @@ static bool pushDepHeight(const DataDep &Dep, const MachineInstr &UseMI,
                                                   Dep.UseOp);
 
   // Update Heights[DefMI] to be the maximum height seen.
-  MIHeightMap::iterator I;
-  bool New;
-  std::tie(I, New) = Heights.insert(std::make_pair(Dep.DefMI, UseHeight));
-  if (New)
+  const auto &[I, Inserted] = Heights.insert({Dep.DefMI, UseHeight});
+  if (Inserted)
```
Comment (s-barannikov) [inline llvm/lib/CodeGen/MachineTraceMetrics.cpp:967]: The coding standard says nothing about structured bindings. But it says something about `auto` -- that using it is discouraged, unless it clearly improves readability.

**Review by s-barannikov — COMMENTED** [review]
The renaming part is fine, but the rest of the changes look more personal prefernce than an attempt to conform to the coding standard.
Feel free to ignore my nitpicks if you get an approval. The history has already been clobbered anyway.

**Comment by artagnon** [comment]
Not pursuing this.

**Pull request closed without merging**